### ISSUE TITLE
TypeError: Cannot read property 'reduce' of undefined

### ISSUE THREAD

**Issue by diegonava6** [johannesjo/super-productivity#391]
META: SP5.3.3 Browser – en – Win32 – Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.97 Safari/537.36

### Steps to Reproduce

1.
2.
3.
4.

### Stacktrace
```
reduce (webpack:///src/app/core/persistence/persistence.service.ts:459:22)
Ur (https://app.super-productivity.com/main.55b23efcec87ac33f036.js:1:1540329)
Ur (https://app.super-productivity.com/main.55b23efcec87ac33f036.js:1:1540434)
Ur (https://app.super-productivity.com/main.55b23efcec87ac33f036.js:1:1539370)
ZoneAwarePromise (webpack:///node_modules/zone.js/dist/zone.js:913:32)
Ur (https://app.super-productivity.com/main.55b23efcec87ac33f036.js:1:1539115)
Ur (https://app.super-productivity.com/main.55b23efcec87ac33f036.js:1:1551191)
remove (webpack:///src/app/core/persistence/persistence.service.ts:449:36)
Ur (https://app.super-productivity.com/main.55b23efcec87ac33f036.js:1:1540329)
Ur (https://app.super-productivity.com/main.55b23efcec87ac33f036.js:1:1540434)
```

**Comment by johannesjo** [comment]
I am wondering how your setup is as I am unable to reproduce this. Have you been using the app before or did you start with a fresh set of data? And do you have any projects created at all?

**Comment by ruanformigoni** [comment]
Hi, I think this problem is the same I'm having:
```text
Error: Uncaught (in promise): TypeError: Cannot read property 'reduce' of undefined TypeError: Cannot read property 'reduce' of undefined at t. (file:///snap/superproductivity/651/resources/app.asar/dist/webpack:/src/app/core/persistence/persistence.service.ts:459:23) at file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1540329 at Object.next (file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1540434) at executor (file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1539370) at new t (file:///snap/superproductivity/651/resources/app.asar/dist/webpack:/node_modules/zone.js/dist/zone.js:913:33) at D (file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1539115) at t._loadForProjectIds (file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1551191) at t. (file:///snap/superproductivity/651/resources/app.asar/dist/webpack:/src/app/core/persistence/persistence.service.ts:449:37) at file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1540329 at Object.next (file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1540434)
! Please copy & report !

Error: Uncaught (in promise): TypeError: Cannot read property 'reduce' of undefined TypeError: Cannot read property 'reduce' of undefined at t. (file:///snap/superproductivity/651/resources/app.asar/dist/webpack:/src/app/core/persistence/persistence.service.ts:459:23) at file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1540329 at Object.next (file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1540434) at executor (file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1539370) at new t (file:///snap/superproductivity/651/resources/app.asar/dist/webpack:/node_modules/zone.js/dist/zone.js:913:33) at D (file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1539115) at t._loadForProjectIds (file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1551191) at t. (file:///snap/superproductivity/651/resources/app.asar/dist/webpack:/src/app/core/persistence/persistence.service.ts:449:37) at file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1540329 at Object.next (file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1540434)
t.<anonymous> (file:///snap/superproductivity/651/resources/app.asar/dist/webpack:/src/app/core/persistence/persistence.service.ts:459:23)
Ur (file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1540329)
Ur (file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1540434)
Ur (file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1539370)
new t (file:///snap/superproductivity/651/resources/app.asar/dist/webpack:/node_modules/zone.js/dist/zone.js:913:33)
Ur (file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1539115)
Ur (file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1551191)
t.<anonymous> (file:///snap/superproductivity/651/resources/app.asar/dist/webpack:/src/app/core/persistence/persistence.service.ts:449:37)
Ur (file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1540329)
Ur (file:///snap/superproductivity/651/resources/app.asar/dist/main.05eca3ac60f95ea85f95.js:1:1540434)
META: SP5.3.4 Electron – en-US – Linux x86_64 – Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) superProductivity/5.3.4 Chrome/80.0.3987.165 Electron/8.3.0 Safari/537.36
```
System information with _neofetch_:
```shell
OS: Ubuntu 20.04 LTS x86_64 
Host: 81V7 Lenovo IdeaPad S145-15API 
Kernel: 5.4.0-33-generic 
Uptime: 2 days, 11 hours, 26 mins 
Packages: 2612 (dpkg), 22 (snap) 
Shell: fish 3.1.0 
Resolution: 1920x1080 
DE: GNOME 
WM: Mutter 
WM Theme: Yaru-dark 
Theme: Yaru-dark [GTK2/3] 
Icons: Yaru [GTK2/3] 
Terminal: x-terminal-emul 
CPU: AMD Ryzen 7 3700U with Radeon Vega Mobile Gfx (8) @ 2.300GHz 
GPU: AMD ATI 03:00.0 Picasso 
Memory: 4392MiB / 5882MiB 
```

**Comment by johannesjo** [comment]
@ruanformigoni thank you. Same question to you: ave you been using the app before or did you start with a fresh set of data? And do you have any projects created at all? When did the error start to occur? Directly after (updating and) starting the app? Or after you did something?

**Comment by ruanformigoni** [comment]
I have been using the app regularly for some time now. I have about 7 projects. Today, just before I set up a novel gitlab integration. Now each time I start up the app it goes to this error screen:

![Error Screen](https://i.imgur.com/DPpWhkb.png)

I'm unable to use it anymore, unfortunatelly.

**Comment by johannesjo** [comment]
Could you by any chance provide me with your latest backup data (can be found under ` `/.config/superProductivity/backups`)? I'd really like to get to the bottom of this, as there seem to be quite a lot of people having the same problem.

**Comment by ruanformigoni** [comment]
E-mail sent. Anything else don't hesitate to ask.

**Comment by johannesjo** [comment]
@ruanformigoni Thank you very much! I am investigating.

**Comment by johannesjo** [comment]
So I am able to just import the newest backup without any issue. This might work as a quick fix (but make sure to backup your complete superProductivity folder before in case something goes wrong)

@ruanformigoni do you by any chance remember the exact steps you did before this error started to occur?

**Comment by ruanformigoni** [comment]
Yes, I opened up the app as usual, I noticed an recurring error message that gitlab was not properly configured, but sometimes this happened even when then configuration is correct. So, afterwards I clicked on new project, inserted the project ref, the gitlab token and the user reference in the last field. I clicked in the save button. The "gitlab config is incorrect" was still showing, so I restarted the app (this usually fixes the problem) and this screen started to appear.

A google drive error also appears shortly in the initial error screen:
![](https://i.imgur.com/bYnFfWp.png)

This are the contents of SUP_STORE:
![](https://i.imgur.com/2QyHsO6.png)
